src/gtfs/read.rs

```diff
@@ -267,14 +268,18 @@ impl<'de> ::serde::Deserialize<'de> for RouteType {
 impl_id!(Route);
 
 impl Route {
-    fn get_line_key(&self) -> (Option<String>, String) {
+    fn get_line_key(&self, read_as_line: bool) -> (Option<String>, String, Option<Uuid>) {
         let name = if self.short_name != "" {
             self.short_name.clone()
         } else {
             self.long_name.clone()
         };
-
-        (self.agency_id.clone(), name)
+        let key = if read_as_line {
+            Some(Uuid::new_v4())
```

Comment:
Is it necessary to use uuid here? Can't you use the index of a route in `gtfs_routes`?

Reply:
This requires passing an additional parameter

Reply:
if you use Idx instead of Uuid

```rust 
type MapLineRoutes<'a> = BTreeMap<(Option<String>, String, Option<Idx<Route>>), Vec<&'a Route>>;

fn map_line_routes<'a>(
    gtfs_routes: &'a CollectionWithId<Route>,
    gtfs_trips: &[Trip],
    read_as_line: bool,
) -> MapLineRoutes<'a> {
    let mut map = BTreeMap::new();
    for (idx, r) in gtfs_routes.iter().filter(|(_, r)| {
        if !gtfs_trips.iter().any(|t| t.route_id == r.id) {
            warn!("Couldn't find trips for route_id {}", r.id);
            return false;
        }
        true
    }) {
        map.entry(r.get_line_key(read_as_line, idx))
            .or_insert_with(Vec::new)
            .push(r);
    }
    map
}
```

I should work, what do you think?

Reply:
doone in f33d5556747a5738d0d81909c89afd2163969329